cryptography/src/commonJvmAndroid/kotlin/com.wire.kalium.cryptography/ProteusClientCoreCryptoImpl.kt

```diff
@@ -218,5 +218,24 @@ class ProteusClientCoreCryptoImpl private constructor(
                 throw ProteusStorageMigrationException("Failed to migrate from crypto box at $rootDir", exception)
             }
         }
+
+        private fun mapProteusExceptionToErrorCode(proteusException: ProteusExceptionNative): ProteusException.Code {
+            return when (proteusException) {
+                is ProteusExceptionNative.SessionNotFound -> ProteusException.Code.SESSION_NOT_FOUND
+                is ProteusExceptionNative.DuplicateMessage -> ProteusException.Code.DUPLICATE_MESSAGE
+                is ProteusExceptionNative.RemoteIdentityChanged -> ProteusException.Code.REMOTE_IDENTITY_CHANGED
+                is ProteusExceptionNative.Other -> ProteusException.fromProteusCode(proteusException.v1.toInt())
+            }
+        }
+
+        @Suppress("MagicNumber")
+        private fun mapProteusExceptionToRawIntErrorCode(proteusException: ProteusExceptionNative): Int {
```

Comment:
Those mappers are here because we've to rely on the  native `ProteusException` type directly, and we're not having the right dependency in the `ProteusException`

Reply:
Going forward I'd suggest aligning `ProteusException` with the `ProteusExceptionNative`. We want to go away from dealing `Int` numbers for error handling.

Doesn't have done right away but maybe create a TODO with a ticket?

Reply:
Makes sense to have it in a different task, there is also some corecryptobox mappings to the `ProteusException` - but indeed would be nice to also clean it up